pulpcore/app/settings.py

```diff
@@ -281,3 +282,21 @@
             "provided: {}".format(unknown_algs)
         )
     )
+for checksum in ALLOWED_CONTENT_CHECKSUMS:
+    # can't import Artifact here so use a direct db connection
+    try:
+        with connection.cursor() as cursor:
+            cursor.execute(f"SELECT count(pulp_id) FROM core_artifact WHERE {checksum} IS NULL")
+            row = cursor.fetchone()
+    except Exception:
+        # our check could fail if the table hasn't been created yet or we can't get a db connection
+        pass
+    else:
+        if row[0] > 0:
+            raise ImproperlyConfigured(
+                _(
+                    "ALLOWED_CONTENT_CHECKSUMS cannot contain checksum '{}' since there is "
+                    "already content in the database and checksums cannot be added to "
+                    "ALLOWED_CONTENT_CHECKSUMS (only removed)."
```

Comment:
```suggestion
                    "ALLOWED_CONTENT_CHECKSUMS cannot contain checksum '{}' since checksums can only be removed and not added to "
                    "ALLOWED_CONTENT_CHECKSUMS in existing deployments."
```

My take: From a user perspective the _why_ should not matter too much. It is simply not allowed.
That is what I would say. Maybe it can be articulated better.